chapter2.adoc

```diff
@@ -31,62 +31,62 @@ A pending request to enter Debug Mode can dynamically change from a disallowed s
+The `prv` and `v` fields in `dcsr` are updated with the current privilege mode on Debug Mode entry, and are used to set the new privilege mode on resume from Debug Mode.  The debugger may modify the `prv` and `v` fields, to alter the mode of execution upon resume, but the allowed values are constrained by the debugger privilege. 
+
+The maximum debug privilege level that can be configured in `prv` and `v` is shown in <<maxdbgpriv>>. On a write that attempts to write an illegal value to `prv` and/or `v`, the fields retain legal values. Illegal privilege levels include values higher than the maximum allowed debug privilege.
 
 [[maxdbgpriv]]
 [options="header"]
-.Determining maximum debug access privilege with mdbgen and sdedbgalw
+.Maximum Allowed Resume Privilege Mode
 |=========================================
-| mdbgen | sdedbgalw | Maximum debug privilege allowed 
-| 1      | Don't care      | M                 
-| 0      | 1      | S(HS)             
-| 0      | 0      | None               
+| mdbgen | sdedbgalw | Maximum privilege allowed on resume
+| 1      | Don't care      | M
+| 0      | 1      | S(HS)
+| 0      | 0      | None
 |=========================================
 
 [NOTE]
 As the `prv` and `v` fields are Write Any Read Legal (WARL) fields, the external debugger is able to read back the written value to determine the maximum debug privilege level.  
 
 ==== Privilege Level Changing Instructions
 
-The RISC-V Debug Specification cite:[dbgspec] defines that the instructions that change the privilege mode have UNSPECIFIED behavior when executed within the Program Buffer, with exception of the ebreak instruction. In Sdsec, those instructions including mret, sret, uret, ecall, must either act as NOP or trigger an exception (stopping execution and setting `cmderr` to 3) in Program Buffer. Notably, these instructions retain their normal functionality during single stepping.
+The RISC-V Debug Specification cite:[dbgspec] defines that the instructions that change the privilege mode have UNSPECIFIED behavior when executed within the Program Buffer, with exception of the EBREAK instruction. In Sdsec, privilege changing instructions (other than EBREAK) executed in the Program Buffer must either act as a NOP or raise an exception (stopping execution and setting `abstractcs.cmderr` to 3).
 
 ==== Interrupt during Single Stepping
 
-The interrupt can be disabled by `stepie` in `dcsr` during single stepping. When `mdbgen` is 1, `stepie` disables interrupts in all privilege modes for the hart. When `mdbgen` is 0 and `sdedbgalw` is 1, only interrupts delegated to the supervisor domain are disabled, while interrupts that trap to M-mode are not affected.
+Interrupts during single-step can be disabled by setting `dcsr.stepie`=1. When `mdbgen` is 1, `stepie` disables interrupts in all privilege modes for the hart. When `mdbgen` is 0 and `sdedbgalw` is 1, only delegated interrupts are disabled, while interrupts that trap to M-mode are not affected.
```

Comment:
Hi @bcstrongx ,
Interrupts during single-step should be disabled by setting `dcsr.stepie`=0, not 1.
Is it typo ?

Reply:
Yes, thanks, should be `dcsr.stepie`=0.

Reply:
Fixed in https://github.com/riscv-non-isa/riscv-external-debug-security/pull/68